impexp-client/src/main/java/org/citydb/gui/modules/common/filter/XMLQueryView.java

```diff
@@ -463,14 +467,22 @@ public Icon getIcon() {
 
     @Override
     public void loadSettings() {
-        QueryConfig query = config.getExportConfig().getQuery();
-        xmlText.setText(marshalQuery(query, config.getNamespaceFilter()));
+        QueryConfig query = queryConfigSupplier.get();
+        xmlText.setText(marshalQuery(query, configNamespaceFilterSupplier.get()));
     }
 
     @Override
     public void setSettings() {
         QueryConfig query = unmarshalQuery();
-        config.getExportConfig().setQuery(query);
+        queryConfigSupplier.get().setTargetSrs(query.getTargetSrs());
+        queryConfigSupplier.get().setFeatureTypeFilter(query.getFeatureTypeFilter());
+        queryConfigSupplier.get().setAppearanceFilter(query.getAppearanceFilter());
+        queryConfigSupplier.get().setCounterFilter(query.getCounterFilter());
+        queryConfigSupplier.get().setLodFilter(query.getLodFilter());
+        queryConfigSupplier.get().setProjectionFilter(query.getProjectionFilter());
+        queryConfigSupplier.get().setSelectionFilter(query.getSelectionFilter());
+        queryConfigSupplier.get().setSorting(query.getSorting());
+        queryConfigSupplier.get().setTiling(query.getTiling());
```

Comment:
Maybe we should add a `copyFrom(QueryConfig other)` to the `QueryConfig` class? Then we could replace lines 477-485 with one call `queryConfigSupplier.get().copyFrom(query)`.

The advantage is that the `QueryConfig` class must make sure that all elements of a query are correctly copied. For example, if we extend the `QueryConfig` class in future, we only need to adapt the `copyFrom` method to account for these changes.

Reply:
Good idea

Reply:
Done

Reply:
Thanks. Well, this change was indeed required. Otherwise, this line `query.setLocalProperty("unmarshallingFailed", true);` (line 383) would have been without effect. It now works again because your `copyFrom` method also takes care of the local properties.

Hm, instead of copying we could also require another `@FunctionalInterface` in the constructor which sets the query object. So instead of

`queryConfigSupplier.get().copyFrom(query)`

something like

`queryConfigSetter.set(query)`

Then we take the object as created and avoid further side effects.

Reply:
Yes, this approach looks much cleaner. see here 372d897